generate bottlerocket rpm inventory file to share with host containers

jpculp opened the pull request:
**Description of changes:**

This grants host containers context to the underlying Bottlerocket OS packages by generating a JSON-formatted inventory file at the time of RPM install and mounting it as read-only to `/var/lib/bottlerocket/inventory/application.json`.

**Testing done:**

- Built `aws-k8s-1.21` instance.
- `cat` mounted file in both the control and admin host containers.

**Example output:**

https://gist.github.com/jpculp/c1aa3142ce8917eeec47fc170f74873d

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

jpculp committed:
packages: add URL to Bottlerocket packages

jpculp committed:
build: inventory installed packages

This adds logic to `rpm2img` to generate an inventory of the installed
rpms to `/usr/share/bottlerocket/application-inventory.json`.

cbgbt commented:
Do you mind sharing an example inventory file as a gist?

bcressey commented on tools/rpm2img:
```diff
@@ -135,7 +135,34 @@ if [ "${PARTITION_PLAN}" == "split" ] ; then
+mapfile -t installed_rpms <<< "$(rpm -qa --root "${ROOT_MOUNT}" \
+  --queryformat "${INVENTORY_QUERY}")"
+
+# shellcheck disable=SC2001
+if INVENTORY_DATA="$(jq --raw-output . \
+  <<< "$(sed "s/bottlerocket-${ARCH}-//g" <<< "${installed_rpms[@]}")" \
+  | jq --slurp 'sort_by(.Name)' | jq '{"Content": .}')"; then
+  printf "%s\n" "${INVENTORY_DATA}" > "${ROOT_MOUNT}/usr/share/bottlerocket/application-inventory.json"
+else
+  echo "something went wrong when generating the package inventory"
+  exit 1
+fi
```
Can we rely on `set -e` instead?

```suggestion
# shellcheck disable=SC2001
INVENTORY_DATA="$(jq --raw-output . \
  <<< "$(sed "s/bottlerocket-${ARCH}-//g" <<< "${installed_rpms[@]}")" \
  | jq --slurp 'sort_by(.Name)' | jq '{"Content": .}')"
printf "%s\n" "${INVENTORY_DATA}" > "${ROOT_MOUNT}/usr/share/bottlerocket/application-inventory.json"
```

I'd also prefer putting the `sed` output into an intermediate variable just to simplify the invocation.

misterek commented:
Out of curiosity, is there any chance that this might be a way to make some progress toward: https://github.com/bottlerocket-os/bottlerocket/issues/848 ?

jpculp commented:
This particular PR is to help support SSM inventories, but the file is generic enough to be used elsewhere. Separately, we are working with AWS Inspector to add support for Bottlerocket.

jpculp committed:
host-ctr: share inventory of installed packages with host containers

jpculp commented:
- Replaced `sed` logic with `jq` sub.
- Split up transforms to make it easier to read (and add future ones).
- Fixed incorrect architecture in cross-compiled builds.
- Removed extraneous error-checking.

cbgbt reviewed (approved):
Nice!
